azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -947,17 +960,18 @@ def _run_op(self, op):
                         self.name, op.name, self.state
                     )
                 )
+                self.state = ReconnectState.LOGICALLY_DISCONNECTED
                 self._clear_reconnect_timer()
                 self._complete_waiting_connect_ops(
                     pipeline_exceptions.OperationCancelled("Explicit disconnect invoked")
                 )
-                self.state = ReconnectState.CONNECTED_OR_DISCONNECTED
                 op.complete()
 
             else:
                 logger.info(
                     "{}({}): State is {}.  Sending op down.".format(self.name, op.name, self.state)
                 )
```

Comment:
Here (and in the above blocks as well) we log the old state right before setting a new state. That's probably going to really confuse up the log statements, I assume it would be better for us to log AFTER adjusting the state